app/controllers/new_administrateur/procedures_controller.rb

```diff
@@ -185,6 +185,12 @@ def transfer
       end
     end
 
+    def expert_list
+      redirect_to admin_procedure_path(@procedure) if !@procedure.allow_expert_review?
+      avis_list = @procedure.dossiers.includes(:avis).map(&:avis).reject(&:empty?)
+      @experts = avis_list.map { |avis| Instructeur.find(avis[0].instructeur_id).user }.uniq
```

Comment:
```suggestion
      if @procedure.allow_expert_review?
        @expert_emails = Avis
          .joins(dossier: :procedure)
          .left_joins(instructeur: :user)
          .where(dossiers: { revision: @procedure.revisions })
          .map(&:email_to_display)
      else
        redirect_to admin_procedure_path(@procedure)
      end
```

- On essaye de limiter les if en fin de ligne pour gagner en lisibilité, et du coup je retire aussi la négation.
- je pense qu'il faut afficher tous les experts ayant reçu une invitation, du coup je retire le `empty?`
- je ne comprends pas trop le chemin que tu as pris (en particulier comment on arrive a `avis[0]`), je propose une requete (**a tester :)**) qui permet de taper directement sur les avis
- le `where(dossiers: { revision: p.revisions })` me semble chelou, je n'ai pas réussi à faire `where(dossiers: { procedure: @procedure)` . Un soucis dans les révisions ? @tchak

Reply:
J'arrive a faire `Avis.joins(dossier: :revision).where(procedure_revisions: { procedure_id: @procedure })`